### PR TITLE
IR-4714 Merge coil setting to engine settings

### PR DESCRIPTION
**Pull request by mshafiqmk**
## Summary

Description: This PR migrates the Coil settings from the **coil-setting** table to the **engine-setting** table. The following changes have been made:

### Migration Script:

Created a new migration script to move existing data from **coil-setting** to **engine-setting**.
The migration script also drops the coil-setting table after the data has been moved.

### Code Update:

Updated **updateAppConfig.ts** to read and write Coil settings from the engine-settings table instead of the coil-setting table.

### Seed File Update:
Updated the **engine-setting** seed file to insert  Coil settings into the engine-settings table.

## References
closes #_insert number here_

### CODE REVIEW

**Review by barankyle — APPROVED** [review]

File: packages/client/scripts/getCoilSettings.ts [hunk 1]
```diff
@@ -42,11 +42,17 @@ export const getCoilSetting = async () => {
 
   const coilSetting = await knexClient
     .select()
-    .from<CoilSettingType>(coilSettingPath)
-    .then(([dbCoil]) => {
-      if (dbCoil) {
-        return dbCoil
+    .from<EngineSettingType>(engineSettingPath)
```
Comment (hanzlamateen) [inline packages/client/scripts/getCoilSettings.ts:45]: We need to update this file to be getEngineSetting and then make it get only those keys which are required from where this method is being called.

**Review by hanzlamateen — CHANGES_REQUESTED** [review]

File: packages/server-core/src/setting/engine-setting/engine-setting.seed.ts [hunk 2]
```diff
@@ -93,3 +84,20 @@ export async function seed(knex: Knex): Promise<void> {
     }
   }
 }
+
+async function generateSeedData(
+  items: { key: string; value: string }[],
+  category: EngineSettingType['category'],
+  type: EngineSettingType['type']
```
Comment (hanzlamateen) [inline packages/server-core/src/setting/engine-setting/engine-setting.seed.ts]: Maybe you can make the default `type` here as private?

**Review by hanzlamateen — CHANGES_REQUESTED** [review]

File: packages/client/scripts/getCoilSettings.ts (hunk 1 above)
Comment (mshafiqmk) [reply packages/client/scripts/getCoilSettings.ts:45]: @hanzlamateen done.

File: packages/server-core/src/setting/engine-setting/engine-setting.seed.ts (hunk 2 above)
Comment (mshafiqmk) [reply packages/server-core/src/setting/engine-setting/engine-setting.seed.ts]: @hanzlamateen done

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**